app/controllers/cart_controller.rb

```diff
@@ -2,8 +2,17 @@ class CartController < ApplicationController
   def add_item
     if current_user == nil || current_user.role != "admin"
       item = Item.find(params[:item_id])
-      cart.add_item(item.id.to_s)
-      flash[:success] = "#{item.name} was successfully added to your cart"
+      item_inventory = item.inventory
+
+      has_inventory = item_inventory > 0 
+      if has_inventory 
+        cart.add_item(item.id.to_s)
+        item.update(inventory: item_inventory - 1)
+        flash[:success] = "#{item.name} was successfully added to your cart"
+      else
+        flash[:error] = "Not enough in inventory"
+      end
+
```

Comment:
This looks super! It is recommended that we avoid nested if statements by utilizing helper methods. Maybe we can throw this in the "refactor if we have time" pile?

Reply:
Ill put that in my notes!